src/payment/strategies/adyenv2/adyenv2-payment-strategy.spec.ts

```diff
@@ -150,6 +150,30 @@ describe('AdyenV2PaymentStrategy', () => {
                 expect(adyenCheckout.create).toHaveBeenCalledTimes(1);
             });
 
+            it('fails mounting scheme payment component', async () => {
+                paymentComponent = getFailingComponent();
+
+                await expect(strategy.initialize(options))
+                    .rejects.toThrow(NotInitializedError);
+            });
+
+            it('fails mounting GiroPay payment component', async () => {
+                jest.spyOn(store.getState().paymentMethods, 'getPaymentMethodOrThrow')
+                    .mockReturnValue(getAdyenV2(AdyenPaymentMethodType.GiroPay));
+
+                paymentComponent = getFailingComponent();
+
+                await expect(strategy.initialize(options))
+                    .rejects.toThrow(NotInitializedError);
+            });
+
+            it('fails mounting card verification component', async () => {
+                cardVerificationComponent = getFailingComponent();
+
+                await expect(strategy.initialize(options))
+                    .rejects.toThrow(NotInitializedError);
+            });
```

Comment:
This test and this one => https://github.com/bigcommerce/checkout-sdk-js/pull/874/files#diff-874c6669fffb1e8c18b7b0bbf4738a5aR153-R158 seems to be the same, you're using the same beforeEach and the same Component `getFailingComponent` and also you're expecting the same error, so maybe these two tests can be converted into a single one. 🍹

Reply:
Actually, they are not testing the same failing component, one is testing cardVerificationComponent and the other one paymentComponent.

Reply:
I still don't see how they are different? the code blocks are identical except for the variable where you assign the result to, which seems unrequired.

```
it('fails mounting card verification component', async () => {
  getFailingComponent();

  await expect(strategy.initialize(options))
    .rejects.toThrow(NotInitializedError);
});

Reply:
Hi Luis, 

First test covers catch block when exception it thrown while mounting `paymentComponent`

![image](https://user-images.githubusercontent.com/8570490/82562456-db6d4000-9b3a-11ea-89c7-c0673673086b.png)

Second test covers catch block when exception it thrown while mounting `cardVerificationComponent`

![image](https://user-images.githubusercontent.com/8570490/82562626-3b63e680-9b3b-11ea-8dc0-b4b51c5ce15c.png)

Reply:
What is confusing of these test is that is hard to know what's exactly going on by checking the test itself. Although it might seem that the overall test is simpler and has less duplicated code, it's harder to follow what the test is trying to achieve. Code in general is encouraged to be DRY, but for unit tests a DAMP approach is preferred:
https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests

Ideally, a beforeEach block should be a setup that is the same for each test, and each test shouldn't be affecting the setup.

```js
it('throws exceptions when a faulty adyen payment component is returned', async () => {
  jest.spyOn(adyenCheckout, 'create')
    .mockImplementationOnce(jest.fn((_method, options) => {
      const { onChange } = options;
      handleOnChange = onChange;

      return getFailingComponent();
  }));

  await expect(strategy.initialize(options))
    .rejects.toThrow(NotInitializedError);
});
```

Reply:
Also, is it possible that `adyenCheckout.create()` returns a null component? Is that closer to reality than having an object with a `mount` function that throws?

```js
jest.spyOn(adyenCheckout, 'create')
    .mockImplementationOnce(jest.fn(() => null));
```